pkg/domain/utils/scp.go

```diff
@@ -1,31 +1,27 @@
 package utils
 
 import (
-	"bytes"
 	"fmt"
 	"io/ioutil"
-	"net"
 	"net/url"
 	"os"
 	"os/exec"
 	"os/user"
 	"strconv"
 	"strings"
-	"time"
-
-	scpD "github.com/dtylman/scp"
 
 	"github.com/containers/common/pkg/config"
+	sshCommon "github.com/containers/common/pkg/ssh"
+
 	"github.com/containers/podman/v4/libpod/define"
 	"github.com/containers/podman/v4/pkg/domain/entities"
-	"github.com/containers/podman/v4/pkg/terminal"
 	"github.com/docker/distribution/reference"
 	"github.com/sirupsen/logrus"
-	"golang.org/x/crypto/ssh"
-	"golang.org/x/crypto/ssh/agent"
 )
 
-func ExecuteTransfer(src, dst string, parentFlags []string, quiet bool) (*entities.ImageLoadReport, *entities.ImageScpOptions, *entities.ImageScpOptions, []string, error) {
+var sshEngine sshCommon.EngineMode
```

Comment:
Package-global variables make nervous as it means shared state for the system-service.  Can you drop a comment why it's needed?  Or maybe you see a away around it.

Reply:
I thought the package variable made the most sense since otherwise I would have to funnel it down from `ExecuteTransfer` to a few functions. I didn't even try the funnel but it might be small enough that I can do it that way

Reply:
yep, easy enough. I will remove the package level variable